custom/ilsgateway/balance.py

```diff
@@ -0,0 +1,154 @@
+from django.core.exceptions import ValidationError
+from django.core.validators import validate_email
+from django.db import transaction
+
+from corehq.apps.locations.models import SQLLocation
+from corehq.apps.sms.mixin import apply_leniency, VerifiedNumber
+from corehq.apps.users.models import WebUser, CommCareUser
+from custom.ilsgateway.models import ILSMigrationStats, ILSMigrationProblem
+from custom.logistics.mixin import UserMigrationMixin
+from custom.logistics.utils import iterate_over_api_objects
+
+
+class BalanceMigration(UserMigrationMixin):
+
+    def __init__(self, domain, endpoint):
+        self.domain = domain
+        self.endpoint = endpoint
+
+    def _get_total_counts(self, func, limit=1, **kwargs):
+        meta, _ = func(limit=limit, **kwargs)
+        return meta['total_count'] if meta else 0
+
+    @transaction.atomic
+    def validate_sms_users(self, date=None):
+        for sms_user in iterate_over_api_objects(
+                self.endpoint.get_smsusers, filters=dict(date_updated__gte=date)
```

Comment:
we should check all sms users every time in case the verified numbers got reassigned